cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -278,29 +278,47 @@ func (g *gitHubRepository) setClientToken(token string) {
 func (g *gitHubRepository) getVersions() ([]string, error) {
 	client := g.getClient()
 
-	// get all the releases
+	// Get all the releases.
 	// NB. currently Github API does not support result ordering, so it not possible to limit results
-	var releases []*github.RepositoryRelease
+	var allReleases []*github.RepositoryRelease
 	var retryError error
 	_ = wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
 		var listReleasesErr error
-		releases, _, listReleasesErr = client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
+		// Get the first page of GitHub releases.
+		releases, response, listReleasesErr := client.Repositories.ListReleases(context.TODO(), g.owner, g.repository, nil)
```

Comment:
Looks like the option to set `PerPage: 100` was dropped during the rebase. Can you please add it back. We would want to keep the number of API calls as small as possible.

Reply:
Ah my bad, adding it back